node/src/components/storage.rs

```diff
@@ -351,14 +354,33 @@ impl Storage {
         Ok(match req {
             StorageRequest::PutBlock { block, responder } => {
                 let mut txn = self.env.begin_rw_txn()?;
-                let outcome = txn.put_value(self.block_db, block.hash(), &block, true)?;
+                if !txn.put_value(
+                    // It's not really appropriate to put block bodes in the metadata database,
+                    // but we have too many databases in storage and we can't make any more.
+
+                    // We can't put this in the block_header_db because we scan it to make our
+                    // indices at startup, so everything in that database must be a block header.
+                    self.block_metadata_db,
```

Comment:
Aren't we overwriting finality signatures with this?

Reply:
Yes, we are. We can't do that.

Reply:
Fixed in 8039fb9ccf0e55a813689f9d44c7fb190e1f42b5